numba/targets/arraymath.py

```diff
@@ -3097,6 +3097,27 @@ def diff_impl(a, n=1):
     return diff_impl
 
 
+@overload(np.array_equal)
+def np_array_equal(a, b):
+
+    if not (type_can_asarray(a) and type_can_asarray(b)):
+        raise TypingError('Both arguments to "array_equals" must be array-like')
+
+    if isinstance(a, types.Integer) and isinstance(b, types.Integer):
```

Comment:
Why single out integers? Are `types.Scalar` or `types.Boolean` not the same pattern?

Reply:
Does `types.Scalar` really exists?

Reply:
I probably mean `types.Number` :)